### PR TITLE
Companymodelfeatures

### PR DESCRIPTION
**Pull request by vijayakarumudi**
Closes AgileVentures/MetPlus_tracker#56

### CODE REVIEW

File: spec/factories/companies.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FactoryGirl.define do
+  factory :company do
+    name "Mycompany"
+ein "234567"
```
Comment (joaopapereira) [inline spec/factories/companies.rb]: Please fix this indentation

File: app/models/company.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  has_many :company_people
+  has_many :addresses, as: :location
+
+  validates_presence_of :name, :ein, :website, :phone, :email
+  validates_length_of   :name, maximum: 100
+  validates_length_of   :website, maximum: 200
+  validates :phone, :phone => true
+  validates :email, :email => true
+  validates :website, :website => true
+
```
Comment (joaopapereira) [inline app/models/company.rb]: Missing ein validation, 
validates :ein, ;ein => true

File: app/models/company.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class Company < ActiveRecord::Base
+  has_many :company_people
+  has_many :addresses, as: :location
+
```
Comment (joaopapereira) [inline app/models/company.rb]: Missing connection to Agencies through agency_companies

Comment (joaopapereira) [inline app/models/company.rb]: Missing connection to Jobs

**Comment by joaopapereira** [comment]
Shouldn't agency_companies table be in this PR?

**Comment by joaopapereira** [comment]
Please review the previous comments

File: app/models/company.rb (hunk 3 above)
Comment (patmbolger) [inline app/models/company.rb]: Vitaya - this association was missing from the ERD for Company model:

has_and_belongs_to_many :agencies

I will upload a revised version of the ERD later today.

**Comment by vijayakarumudi** [comment]
I fixed the indentation, validation for ein. and I added all the missing associations.

**Pull request closed without merging**